File: backend/Domain/Models/Rent.cs
namespace Domain.Models;

public class Rent
{
    public Guid Id { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public Guid ClientId { get; set; }
    public Guid? RenterId { get; set; }
    public Guid? ReceiverId { get; set; }
    public Guid VehicleId { get; set; }
    public Guid RentStatusId { get; set; }

    public User Client { get; set; } = default!;
    public User Renter { get; set; } = default!;
    public User Receiver { get; set; } = default!;
    public Vehicle Vehicle { get; set; } = default!;
    public RentStatus RentStatus { get; set; } = default!;
}

File: frontend/src/app/models/Rent.ts
import { VehicleForRent } from "./Vehicle";


export interface Rent {
    id: string;
    status: string;
    startDate: Date;
    endDate: Date;
    vehicle: string;
}

export interface RentDetails {
    id: string;
    startDate: Date;
    endDate: Date;
    client: string;
    renter?: string;
    receiver?: string;
    vehicle: VehicleForRent;
    status: string;
}

export interface RentCreate {
    vehicleId: string;
    startDate: Date;
    endDate: Date;
}

export interface RentCost {
    regularPrice: number;
    penaltyPrice?: number;
}